app/main/posts/views/post-filters.service.js

```diff
@@ -18,19 +18,73 @@ function PostFiltersService(_, FormEndpoint, TagEndpoint, $q) {
         getQueryParams: getQueryParams,
         getFilters: getFilters,
         setFilters: setFilters,
+        setFilter: setFilter,
         clearFilters: clearFilters,
         clearFilter: clearFilter,
+        clearFilterFromArray: clearFilterFromArray,
         hasFilters: hasFilters,
         getActiveFilters: getActiveFilters,
-        getCleanActiveFilters: getCleanActiveFilters,
+        getUIActiveFilters: getUIActiveFilters,
         setMode: setMode,
         getMode: getMode,
         getModeId: getModeId,
+        getModeEntity: getModeEntity,
         countFilters: countFilters,
+        cleanRemovedValuesFromObject: cleanRemovedValuesFromObject,
+        addIfCurrentObjectMatchesOriginal: addIfCurrentObjectMatchesOriginal,
         reactiveFilters: true,
         qEnabled: false
     };
 
+    /**
+     *
+     *  original = {status: ['draft', 'review', 'archive']};
+     *  current = {status: ['draft', 'review']}
+     *  uiFiltersCurrent = {status: ['draft', 'review', 'archive']}
+     *  Original has 3 filters, then 1 was removed, so current has 2. uiFiltersCurrent has 3 again because the user re-added it.
+     *  We need to compare  original and uiFIltersCurrent.
+     *  If `uiFiltersCurrent` has values that are present in  `original`,
+     *  we need to add them to `current` (*some other function will remove them from uiFiltersCurrent too)
+     *
+     * @param current
+     * @param original
+     */
+    function addIfCurrentObjectMatchesOriginal(currentSS, originalSS, uiObj) {
```

Comment:
fix this signature for something less terrible

Reply:
### Issue found here: 
Array filters (status, categories, surveys) are always added to the saved search hierarchy if we select a saved search. 

- Select a saved search with  some surveys
-  Add another survey category 
- Expected behavior: It should be added outside of the saved search as other new filters, but it's added to the saved search categories instead.